src/lib/shopping-analytics.js

```diff
@@ -25,6 +27,8 @@ function initGenericEventPublisher(config : Config) : Object {
   const convertEvent = eventToFptiConverters(config).eventToFpti;
   return {
     publishEvent: (event : EventType, payload : Object) => {
+      const prevPayload = payload || {};
+      payload = { ...prevPayload, ...config.shoppingAttributes };
```

Comment:
Overriding payload component you modifying original `payload` object. Object is passing here `as a reference`. 
I don't think it is your intentional behavior.

It should work fine in this case
```
const extendedPayload = {...payload, ...config.shoppingAttributes}
```

Reply:
I agree, ideally object should be immutable. I'll make a change